helix/machine_learning/models/BRNN_base_class.py

```diff
@@ -0,0 +1,137 @@
+        self.model_ = nn.Sequential(*layers)
+
+        for layer in self.model_:
+            if isinstance(layer, nn.Linear):
+                nn.init.xavier_uniform_(layer.weight)
+
+        self.num_params_ = sum(p.numel() for p in self.model_.parameters())
+
+    def _get_activation(self, activation):
+        activations = {
+            ActivationFunctions.ReLU: nn.ReLU(),
+            ActivationFunctions.Tanh: nn.Tanh(),
+            ActivationFunctions.Sigmoid: nn.Sigmoid(),
+            ActivationFunctions.LeakyReLU: nn.LeakyReLU(),
+        }
+        if activation not in activations:
+            raise ValueError("Unsupported activation function")
+        return activations[activation]
+
+    def _make_loss(self, problem_type):
+        if problem_type == ProblemTypes.Regression:
+            self.loss = nn.MSELoss()
+        elif problem_type == ProblemTypes.Classification:
+            self.loss = nn.CrossEntropyLoss()
+        else:
+            raise ValueError(f"Unsupported problem type: {problem_type}")
+
+    def _prepare_data(self, X, y):
+
+        X, y = check_X_y(X, y, multi_output=False)
+        self.n_features_in_, self.n_samples_ = X.shape[1], X.shape[0]
+        y = y.reshape(-1, 1) if len(y.shape) == 1 else y
+
+        self.x_scaler_ = StandardScaler()
+        self.y_scaler_ = StandardScaler()
+
+        X = self.x_scaler_.fit_transform(X)
+        y = self.y_scaler_.fit_transform(y)
+
+        X_tensor = torch.tensor(X, dtype=torch.float32)
+        y_tensor = torch.tensor(y, dtype=torch.float32)
+
+        return X_tensor, y_tensor
+
+    def fit(self):
+        raise NotImplementedError
+
+    def _get_eigenvalues(self, X_tensor, y_tensor):
+        """Computes eigenvalues using Fisher Information Matrix (FIM) instead of Hessian."""
+        outputs = self.model_(X_tensor)
+        loss = self.loss(outputs, y_tensor)
+
+        params = list(self.model_.parameters())
+        grads = torch.autograd.grad(loss, params, create_graph=True)
+        fisher_matrix = torch.zeros(
+            sum(p.numel() for p in params), sum(p.numel() for p in params)
+        )
+
+        grad_flat = torch.cat([g.view(-1) for g in grads])
+        for i in range(len(grad_flat)):
+            second_grads = torch.autograd.grad(grad_flat[i], params, retain_graph=True)
+            fisher_matrix[i] = torch.cat([sg.view(-1) for sg in second_grads])
+
+        eigenvalues = torch.linalg.eigvalsh(fisher_matrix).detach().numpy()
+        return eigenvalues
+
+    def _calculate_evidence(self, alpha, beta, eigenvalues, error, reg_term):
+        N = len(eigenvalues)
+        return (
+            -0.5 * (beta * error + alpha * reg_term)
+            - 0.5 * np.sum(np.log(alpha + beta * eigenvalues))
+            + 0.5 * (N * np.log(alpha) + self.n_samples_ * np.log(beta))
+        )
+
+    def predict(self):
+        raise NotImplementedError
+
+    def predict_proba(self):
+        raise NotImplementedError
```

Comment:
This gets provided by the `ClassifierMixin` class in the your classifier class, so you shouldn't need this here. You still need to override it in the classifier class though